.github/workflows/ci.yml

```diff
@@ -1,8 +1,8 @@
 name: 'CI Pipeline'
 on:
-  pull_request:
+  push:
 jobs:
-  pr:
+  install:
```

Comment:
Install job - run a matrix of os's and install then cache all node_modules. This will precede any other run and we can restore this from cache later, that way we dont install node modules for every single job / step that requires the same dependencies.